querybook/server/logic/data_element.py

```diff
@@ -24,12 +25,32 @@ def get_data_element_by_name(name: str, session=None):
 
 
 @with_session
-def get_data_element_associations_by_column_id(column_id: int, session=None) -> dict:
-    return (
+def get_data_element_association_by_column_id(
+    column_id: int, session=None
+) -> DataElementAssociationDict:
+    associations = (
         session.query(DataElementAssociation)
         .filter(DataElementAssociation.column_id == column_id)
         .all()
     )
+    if not associations:
+        return None
+
+    # check if there are more than 1 association type
+    association_types = set([r.type for r in associations])
+    if len(association_types) > 1:
+        LOG.error(
+            f"Column {column_id} has more than one data element associated with it"
+        )
+        return None
+
+    data_element = {}
+    for row in associations:
+        data_element["type"] = row.type.value
+        data_element[row.property_name] = (
+            row.data_element.to_dict() if row.data_element else row.primitive_type
```

Comment:
don't serialize it here. Let it be serialized automatically when returning

Reply:
api json encoder wont do automatic serialization for nested objects.